pypeit/core/flat.py

```diff
@@ -493,24 +508,28 @@ def flatfield(sciframe, flatframe, bpix, illum_flat=None, snframe=None, varframe
 
     # New image
     retframe = np.zeros_like(sciframe)
-    w = np.where(final_flat > 0.0)
-    retframe[w] = sciframe[w]/final_flat[w]
-    if w[0].size != final_flat.size:
-        ww = np.where(final_flat <= 0.0)
-        bpix[ww] = 1.0
-    # Variance?
-    if varframe is not None:
-        # This is risky -- Be sure your flat is well behaved!!
-        retvar = np.zeros_like(sciframe)
-        retvar[w] = varframe[w]/final_flat[w]**2
-        return retframe, retvar
-    # Error image
-    if snframe is None:
+#    w = np.where(final_flat > 0.0)
+    # TODO: use utils.inverse() instead?
+    indx = final_flat > 0.0
+    retframe[indx] = sciframe[indx]/final_flat[indx]
+    if bpm is not None and np.sum(indx) != final_flat.size:
```

Comment:
For np.sum(indx) != final_flat.size, I think not np.all(indx) is equivalent and easier to read. It also does not require doing a sum, since the first False encountered will trigger it to be False.

Reply:
Good call!  Changed.